### PR TITLE
2.28 only: tests/test_suite_ecp: Fix ECP group compare test

### PR DESCRIPTION
**Pull request by gowthamsk-arm**
## Description

ECP group compare function should not check the value of T. We only need to assert the value of T after the ECP group copy function is called.

Relates to #6707.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog**: Not needed as this is a test data
- [x] **backport**: This applies only to mbedtls-2.28
- [x] **tests**: Provided

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

### CODE REVIEW

**Review by mpg — DISMISSED** [review]
LGTM, thanks!

**Review by tgonzalezorlandoarm — DISMISSED** [review]
LGTM :)

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Sorry, this still doesn't look right.

File: tests/suites/test_suite_ecp.function
Comment (gowthamsk-arm) [reply tests/suites/test_suite_ecp.function:64]: Yes, it does. I missed T_size here. Will update it and add a comment. :)

**Review by tgonzalezorlandoarm — APPROVED** [review]
LGTM

**Comment by gilles-peskine-arm** [comment]
The task requires both a PR to development and a PR to 2.28, but they're going to be very different, so there's no reason to wait until development is ready, we can merge the 2.28 PR independently.

**Pull request merged**
